service/history/workflow/mutable_state_impl.go

```diff
@@ -2946,6 +3051,67 @@ func (e *MutableStateImpl) ReplicateWorkflowExecutionSignaled(
 	return nil
 }
 
+// FIXME: name this better
+func (e *MutableStateImpl) NewWorkflowForRetryOrCron(
+	firstEventID int64,
+	startAttr *historypb.WorkflowExecutionStartedEventAttributes,
+	lastCompletionResult *commonpb.Payloads,
+	failure *failurepb.Failure,
+	backoffInterval time.Duration,
+	newType NewWorkflowType,
```

Comment:
the `newType NewWorkflowType` is only used to determine the continue continue as new initiator, so why not pass in the continue as new initiator?

https://github.com/temporalio/temporal/pull/1745/files#diff-8b569e7d448b915cfc54cba06dfa7dfa8d639218eab08286169ede455af3dd10R1270

Reply:
they were more different in an earlier version of the diff.

so, the whole idea of passing in a "continue as new initiator" for something that isn't continue-as-new (anymore) is sort of weird. do we need to do it at all? I guess it's potentially useful information, just named badly....